src/Yarhl/IO/DataStream.cs

```diff
@@ -497,19 +492,102 @@ public void WriteTo(DataStream stream)
                 throw new ObjectDisposedException(nameof(stream));
 
             long currPos = Position;
-            Seek(0, SeekMode.Start);
+            Seek(start, SeekMode.Start);
 
             const int BufferSize = 70 * 1024;
             byte[] buffer = new byte[Length > BufferSize ? BufferSize : Length];
 
-            while (!EndOfStream) {
+            while (!EndOfStream)
+            {
                 int read = BlockRead(this, buffer);
                 stream.Write(buffer, 0, read);
             }
 
             Seek(currPos, SeekMode.Start);
         }
 
+        /// <summary>
+        /// Writes a defined length stream into another DataStream starting in a defined position.
+        /// </summary>
+        /// <param name="start">Defined starting position.</param>
+        /// <param name="length">Defined length to be written.</param>
+        /// <param name="stream">Output DataStream.</param>
+        public void WriteSegmentTo(long start, int length, DataStream stream)
```

Comment:
Length should be a `long` value to be consistent with the type of the property `Length`.

Reply:
Done